### PR TITLE
PRO-4174: Deceased Alias

### PR DESCRIPTION
**Pull request by Wivenhoe-HMCTS**
add deceased alias to the caveat journey

### CODE REVIEW

File: app/steps/ui/deceased/alias/index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        };
+        return nextStepOptions;
+    }
+
+    getContextData(req) {
+        const ctx = super.getContextData(req);
+        const formdata = req.session.form;
+        ctx.deceasedName = FormatName.format(formdata.deceased);
+        return ctx;
+    }
```
Comment (LucaDelBuonoHMCTS) [inline app/steps/ui/deceased/alias/index.js]: Are you not using the same replacement method you said David wants to use?

File: app/steps/ui/deceased/alias/template.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{% extends "includes/two_thirds_form.html" %}
+
+{% from "widgets/textfields.html" import textField %}
+{% from "widgets/radiobuttons.html" import optionsRadio %}
+
+{% block question %}
+    <h1 class="form-title heading-large">{{content.question | replace('{deceasedName}', fields.deceasedName.value) | safe }}</h1>
```
Comment (LucaDelBuonoHMCTS) [inline app/steps/ui/deceased/alias/template.html]: Change this accordingly if you amend the method to replace the deceased name.

File: app/steps/ui/deceased/otherNames/index.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    getContextData(req) {
+        const ctx = super.getContextData(req);
+        if (!ctx.otherNames) {
+            set(ctx, 'otherNames.name_0.firstName', '');
+            set(ctx, 'otherNames.name_0.lastName', '');
+        }
+        const formdata = req.session.form;
+        ctx.deceasedFullName = FormatName.format(formdata.deceased);
+        return ctx;
+    }
```
Comment (LucaDelBuonoHMCTS) [inline app/steps/ui/deceased/otherNames/index.js]: Same thing I guess?

File: app/steps/ui/deceased/otherNames/template.html [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                <li><a href="#otherNames_{{index}}_{{error.param}}">{{error.msg.summary | safe}}</a></li>
+            {% endfor %}
+        {% endfor %}
+        </ul>
+    </div>
+    {% endif %}
+{% endblock %}
+
+{% block question %}
+    <h1 class="form-title heading-large">{{content.question | replace('{deceasedName}', fields.deceasedFullName.value) | safe }}</h1>
```
Comment (LucaDelBuonoHMCTS) [inline app/steps/ui/deceased/otherNames/template.html]: As above

File: app/steps/ui/deceased/alias/index.js (hunk 1 above)
Comment (Wivenhoe-HMCTS) [reply app/steps/ui/deceased/alias/index.js]: agreed with David to leave template substitution as PA journey

File: app/steps/ui/deceased/alias/template.html (hunk 2 above)
Comment (Wivenhoe-HMCTS) [reply app/steps/ui/deceased/alias/template.html]: agreed with David to leave template substitution as PA journey

File: app/steps/ui/deceased/otherNames/index.js (hunk 3 above)
Comment (Wivenhoe-HMCTS) [reply app/steps/ui/deceased/otherNames/index.js]: agreed with David to leave template substitution as PA journey

File: app/steps/ui/deceased/otherNames/template.html (hunk 4 above)
Comment (Wivenhoe-HMCTS) [reply app/steps/ui/deceased/otherNames/template.html]: agreed with David to leave template substitution as PA journey

File: test/component/deceased/testAlias.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        });
+
+        it(`test it redirects to deceased other names page: ${expectedNextUrlForDeceasedOtherNames}`, (done) => {
+            const data = {
+                alias: 'Yes'
+            };
+            testWrapper.testRedirect(done, data, expectedNextUrlForDeceasedOtherNames);
+        });
+
+        it(`test it redirects to deceased other names page: ${expectedNextUrlForEndOfJourney}`, (done) => {
```
Comment (LucaDelBuonoHMCTS) [inline test/component/deceased/testAlias.js]: Update the test description

Comment (Wivenhoe-HMCTS) [reply test/component/deceased/testAlias.js]: changes description to:
 test it redirects to end of journey page:

**Pull request merged**